src/webgpu/web_platform/external_texture/video.spec.ts

```diff
@@ -151,26 +151,105 @@ Test that importing different video formats into an external texture and samplin
           exp: new Uint8Array([0x00, 0xff, 0x00, 0xff]),
         }
       );
+    });
+  });
 
-      // This function submits the same render pass as above, but should result in an error because the
-      // GPUExternalTexture will be destroyed via microtask upon requestAnimationFrame.
-      const ensureDestroyed = function () {
-        const commandEncoder2 = t.device.createCommandEncoder();
-        const passEncoder2 = commandEncoder2.beginRenderPass({
-          colorAttachments: [
+g.test('importExternalTexture,destroy')
+  .desc(
+    `
+Test that GPUExternalTextures are destroyed with a microtask.
+`
+  )
+  .fn(async t => {
+    const videoUrl = getResourcePath('red-green.webmvp8.webm');
+    const video = document.createElement('video');
+    video.src = videoUrl;
+
+    await startPlayingAndWaitForVideo(video, () => {
+      const colorAttachment = t.device.createTexture({
+        format: kFormat,
+        size: { width: kWidth, height: kHeight, depthOrArrayLayers: 1 },
+        usage: GPUTextureUsage.COPY_SRC | GPUTextureUsage.RENDER_ATTACHMENT,
+      });
+
+      const pipeline = t.device.createRenderPipeline({
```

Comment:
Are all these helper objects (pipeline, bind group, etc.) copied from above? They're not really material to this test, so maybe they can be split into a helper function and shared.

Reply:
Added GetPipeline, GetBG helpers.